core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -661,8 +661,7 @@ open class TileInfo {
 
         if (resource != null && !ruleset.tileResources.containsKey(resource)) resource = null
         if (resource != null) {
-            val resourceObject = ruleset.tileResources[resource]!!
-            if (resourceObject.terrainsCanBeFoundOn.none { it == baseTerrain || terrainFeatures.contains(it) })
+            if (ruleset.tileResources[resource]!!.mustRemoveFromTile(this))
```

Comment:
Why not just !isAllowedOnTile(this)? Why make an extra function? I think the performance difference is negligible

Reply:
There's a comment to that (is there)? The original code did already check differently, so I left the discrepancy but wrapped it.

Reply:
I think this should be the same check, if we're unifying things